### PR TITLE
Feature/sc 7680/change postgres connector

### PR DESCRIPTION
**Pull request by CeEv**
# Description

## Links to Tickets or other pull requests
- https://ticketsystem.schul-cloud.org/browse/SC-7680

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/queries/events/alarms/deleteAlarms.js [hunk 1]
```diff
@@ -1,21 +1,11 @@
-const getClient = require('../../../infrastructure/database');
-const errorMessage = require('../../utils/errorMessage');
+const getClient = require('../../../infrastructure/databasePromise');
```
Comment (alriese) [inline src/queries/events/alarms/deleteAlarms.js:1]:
```suggestion
const db = require('../../../infrastructure/databasePromise');
```

File: src/queries/events/deleteEvents.js [hunk 2]
```diff
@@ -1,25 +1,11 @@
...
-            if (error) {
-                errorMessage(query, error);
-                reject(error);
-            } else {
-                resolve(result.rows);
-            }
-        });
-    });
+async function deleteEvents(eventId, scopeIds) {
+	let query = `DELETE FROM events WHERE event_id = $1 AND scope_id = ANY($2) RETURNING ${allColumns}`;
```
Comment (alriese) [inline src/queries/events/deleteEvents.js:5]:
```suggestion
	const query = `DELETE FROM events WHERE event_id = $1 AND scope_id = ANY($2) RETURNING ${allColumns}`;
```

**Review by Metauriel — APPROVED** [review]

File: src/queries/events/deleteEvents.js [hunk 3]
```diff
@@ -1,25 +1,11 @@
...
+	let params = [eventId, scopeIds];
```
Comment (alriese) [inline src/queries/events/deleteEvents.js:6]:
```suggestion
	const params = [eventId, scopeIds];
```

File: src/queries/events/getRawEvents.js [hunk 4]
```diff
@@ -11,55 +11,86 @@ const {
...
+	startAfterAndEndAfter
+	fromDate = 3
+	untilDate = 5
+	2 < 5 && 4 > 3
+	-------------
+	dtstart = 2
+	dtend = 4
+	
+	`dtstart < untilDate AND dtend > fromDate`;
+*/
```
Comment (alriese) [inline src/queries/events/getRawEvents.js:95]: Do we need these examples or is it sufficient to link to the explanation e.g. on StackOverflow: https://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap

File: src/queries/subscriptions/insertOriginalSubscription.js [hunk 5]
```diff
@@ -1,21 +1,11 @@
...
-                resolve(result.rows[0]);
-            }
-        });
-    });
+async function insertOriginalSubscription(subscriptionId, scopeId) {
+    const query = 'INSERT INTO original_subscriptions '
+        + '(subscription_id, scope_id) '
+        + 'VALUES ($1, $2) '
+        + 'RETURNING subscription_id, scope_id';
+    return db.query(query, [subscriptionId, scopeId]);
```
Comment (alriese) [inline src/queries/subscriptions/insertOriginalSubscription.js:8]:
```suggestion
    const result = await db.query(query, [subscriptionId, scopeId]);
    return result[0];
```

File: src/security/authentication.js [hunk 6]
```diff
@@ -1,56 +1,65 @@
...
+				const err = {
+					message: 'Invalid Authorization token!',
+					status: 401,
+					title: 'Unauthorized',
+				};
+				next(err);
+			});
+	} else {
+		const err = {
+			message: 'Invalid Authorization token!',
```
Comment (alriese) [inline src/security/authentication.js:31]:
```suggestion
			message: 'Missing Authorization token!',
```

**Review by alriese — CHANGES_REQUESTED** [review]

**Review by alriese — APPROVED** [review]

**Review by Metauriel — APPROVED** [review]

**Pull request merged**